File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    stk = [];
    
    for i in arr:
        if not stk: stk.append(i);
        elif stk[-1] == i: stk.pop();
        else: stk.append(i);
        
    return stk if len(stk) > 0 else [-1]

File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    const stk = [arr[0]];
    let i = 1;
    
    while(i < arr.length){
        if(stk[stk.length - 1] === arr[i]) stk.pop();
        else stk.push(arr[i]);
        i++;
    }
    return stk.length === 0 ? [-1] : stk
}